### PR TITLE
add gender extractor

### PR DESCRIPTION
**Pull request by ZwEin27**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from setuptools import Extension,find_packages
+from os import path
+
+setup(
+    name = 'digGenderExtractor',
+    version = '0.1.0',
+    description = 'digGenderExtractor',
+    author = 'Lingzhe Teng',
+    author_email = 'zwein27@gmail.com',
+    url = 'https://github.com/ZwEin27/dig-gender-extractor',
```
Comment (jasonslepicka) [inline setup.py]: These should be the usc-isi-i2 urls

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    download_url = 'https://github.com/ZwEin27/dig-gender-extractor',
+    packages = find_packages(),
+    keywords = ['gender', 'extractor'],
+    install_requires=['digSparkUtil', 'digExtractor']
```
Comment (jasonslepicka) [inline setup.py]: digSparkUtil should not be required.

File: digGenderExtractor/gender_extractor.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return copy.copy(self.metadata)
+
+    def set_metadata(self, metadata):
+        self.metadata = metadata
+        return self
+
+    def get_renamed_input_fields(self):
+        return self.renamed_input_fields
+
+    def set_renamed_input_fields(self, renamed_input_fields):
```
Comment (jasonslepicka) [inline digGenderExtractor/gender_extractor.py]: This can be removed because it's inherited

File: digGenderExtractor/gender_extractor.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+from digExtractor.extractor import Extractor
+from digGenderExtractor.digge import DIGGE
+
+class GenderExtractor(Extractor):
+
+    def __init__(self):
+        self.renamed_input_fields = ['text']  # ? renamed_input_fields
+
+    def extract(self, doc):
+        return DIGGE.extract(doc['text'])
```
Comment (jasonslepicka) [inline digGenderExtractor/gender_extractor.py]: We should check sure the 'text' field is defined

File: digGenderExtractor/digge.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        
+        if GE_GENDER_NAME_TRANSGENDER in pap.keys():
+            return GE_GENDER_NAME_TRANSGENDER
+        elif GE_GENDER_NAME_MALE in pap.keys():
+            return GE_GENDER_NAME_MALE
+        else:
+            return GE_GENDER_NAME_FEMALE
+
+if __name__ == '__main__':
+    text = 'female ts male gender'
```
Comment (jasonslepicka) [inline digGenderExtractor/digge.py]: This can be removed

**Review by jasonslepicka — CHANGES_REQUESTED** [review]
Looks good.  Just needs a little cleanup.

File: digGenderExtractor/gender_extractor.py (hunk 4 above)
Comment (ZwEin27) [reply digGenderExtractor/gender_extractor.py]: Hi Jason, thanks for pointing out, 👍 . If 'text' filed is not exist, should I just ignore it or throw a exception? thank you

**Review by jasonslepicka — APPROVED** [review]

**Pull request merged**